### PR TITLE
Add thumbnails for image type messages

### PR DESCRIPTION
**Pull request by rrpuente**
Thumbnails support in the chat-window component

### CODE REVIEW

File: src/ng-chat/components/ng-chat-window/ng-chat-window.component.css [hunk 1]
```diff
@@ -203,9 +203,15 @@
...
-@media only screen and (max-width: 581px) { 
+@media only screen and (max-width: 581px) {
     .ng-chat-window
     {
         position: initial;
     }
 }
+
+
+.responsive-image {
```
Comment (rpaschoal) [inline src/ng-chat/components/ng-chat-window/ng-chat-window.component.css:214]: Should this be added inside `.ng-chat-window` @rrpuente ? Something like `.ng-chat-window .image-message`?

File: src/ng-chat/components/ng-chat-window/ng-chat-window.component.html [hunk 2]
```diff
@@ -40,45 +40,51 @@
...
+				</div>
+        <div *ngSwitchCase="MessageType.Image" [ngClass]="{'sent-chat-message-container': message.fromId == userId, 'received-chat-message-container': message.fromId != userId}">
+
+          <img src="{{message.message}}" class="responsive-image" />
+
 					<span *ngIf="showMessageDate && message.dateSent" class="message-sent-date">{{message.dateSent | date:messageDatePipeFormat}}</span>
 				</div>
 				<div *ngSwitchCase="MessageType.File" [ngClass]="{'file-message-container': true, 'received': message.fromId != userId}">
-					<div class="file-message-icon-container">
+					<div class="file-message-icon-container">ø
```
Comment (rpaschoal) [inline src/ng-chat/components/ng-chat-window/ng-chat-window.component.html:53]: Is it just me or is there a `ø` character here on this line?

File: src/ng-chat/components/ng-chat-window/ng-chat-window.component.css (hunk 1 above)
Comment (rrpuente) [reply src/ng-chat/components/ng-chat-window/ng-chat-window.component.css:214]: yes, it is better like that, Im on it.

File: src/ng-chat/components/ng-chat-window/ng-chat-window.component.html (hunk 2 above)
Comment (rrpuente) [reply src/ng-chat/components/ng-chat-window/ng-chat-window.component.html:53]: sure, I have no idea how that character got here

**Comment by rrpuente** [comment]
> Nice one @rrpuente ! Couple comments on things I believe we should address before merging and releasing this one.
> 
> Could you also please add some unit tests to verify the new image message behavior?

Good day @rpaschoal, I've thought about what tests can be done, but all I can think about is to add messages with the type 3 (Images) to the  tests that are already implemented, Can you suggest what tests can be implemented?

**Comment by rpaschoal** [comment]
Thanks for pushing the changes requested @rrpuente 

It would be nice to verify that the image message gets rendered. You might need to use a `TestBed` for that, you can check out the ng-chat-options test for example. Now that the window was recently broken down to a smaller component it might be easier to achieve this.

**Comment by rpaschoal** [comment]
Once we get coverage for these changes I think we'll be ready to merge and release it @rrpuente 🚀

**Comment by rrpuente** [comment]
Good day @rpaschoal,
I've trying for several hours with no luck.
It looks like the chat-component does not get the reference to the window or something like that.
I just tried a similar test than the one you did for the scroll with the aim of creating a chat window with one message, but subject.chatMessages is always undefined, therefore the fixture cannot find the Html object.
this is the code, can you please help?
it('An image message must be rendered ', () => {
    const chattingToUser = new User();
    const event = {
        keyCode: 13
    };
   let message: Message = {
     fromId: 1,
     toId: 99,
     message:  '',
     type: 3
   }
    chattingToUser.id = 99;
    const currentWindow = new Window(chattingToUser, false, false);
    currentWindow.messages.push(message);
    currentWindow.newMessage = "Test";
    subject.onChatInputTyped(event, currentWindow);
    fixture.detectChanges();
    let img = fixture.debugElement.query(By.css('.image-message'));
    let img1 = fixture.debugElement.query(By.css('img'));
  });

If I add the line  subject.window = currentWindow, then I get the error:
Error: The pipe 'emojify' could not be found!

**Comment by rpaschoal** [comment]
Hi @rrpuente , it seems that you're heading to the right direction from what I can see as there is a fixture in place to mount the Angular component.

From the error message it seems it is trying to render it but crashing for not finding the custom ng-chat `emojify` pipe. There are a few custom pipes in ng-chat and this is one.

Just as an idea from the error message, don't you have to register those pipes to the TestBed as depedencies?

File: demo/offline_bot/src/app/demo-adapter.ts [hunk 3]
```diff
@@ -98,19 +99,25 @@ export class DemoAdapter extends ChatAdapter implements IChatGroupAdapter
                 toId: 999,
                 message: "Hi there, just type any message bellow to test this Angular module.",
                 dateSent: new Date()
-            }
+            },
+            {
+              fromId: 1,
+              toId: 999,
+              type: 3,
```
Comment (rpaschoal) [inline demo/offline_bot/src/app/demo-adapter.ts:106]: We could probably have used the imported `MessageType ` instead of using the value `3` directly.

Comment (rrpuente) [reply demo/offline_bot/src/app/demo-adapter.ts:106]: Definitely you are right, I really didn't realize that one, didn't answer before cuz I was out of town. I hope soon I can add something else.

**Pull request merged**